### ISSUE TITLE
Error when listing software updates

### ISSUE THREAD

**Issue by cd1** [kimchi-project/kimchi#663]
When I try to update the system packages, I always see an error message in the section "Software Updates". Currently, there are a few hundred packages ready to be updated on my system which are displayed correctly when I run `sudo dnf update` on a shell.

I'm running Fedora 22, Kimchi 30ceeea32b3913e09cb7082c4d90022a47594d01, Chrome  43.0.2357.81 (64-bit).

## How to reproduce ##

1. Install and launch Kimchi.
2. Log in with an admin user (e.g. `root`).
3. Switch to the tab "Host" and scroll down to the section "Software Updates". Make sure there are packages to be updated on the system.

## Actual behavior ##

After step (3), the following error message is displayed inside the section "Software Updates":

```
An error occurred while retrieving system information.￼ (Retry)
The server encountered an unexpected condition which prevented it from fulfilling the request.
```

And the following error message is printed to the error console:

```
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 217, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 61, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/home/kimchi/kimchi/src/kimchi/control/base.py", line 328, in index
    return self.get(filter_params)
  File "/home/kimchi/kimchi/src/kimchi/control/base.py", line 315, in get
    resources = self._get_resources(flag_filter)
  File "/home/kimchi/kimchi/src/kimchi/control/base.py", line 272, in _get_resources
    idents = get_list(*self.model_args, **flag_filter)
  File "/home/kimchi/kimchi/src/kimchi/model/host.py", line 405, in get_list
    return self.host_swupdate.getUpdates()
  File "/home/kimchi/kimchi/src/kimchi/swupdate.py", line 93, in getUpdates
    self._scanUpdates()
  File "/home/kimchi/kimchi/src/kimchi/swupdate.py", line 77, in _scanUpdates
    for pkg in self._pkg_mnger.getPackagesList():
  File "/home/kimchi/kimchi/src/kimchi/swupdate.py", line 170, in getPackagesList
    for pkg in self._pkgs:
TypeError: 'NoneType' object is not iterable
```

## Expected behavior ##

After step (3), the packages to be updated should be displayed correctly inside the section "Software Updates".

**Comment by alinefm** [comment]
Is it duplicated of #660 ?

**Comment by cd1** [comment]
No, this is a different issue (even though they're very similar, but with different symptoms and different reasons). Actually, when I try to reproduce this issue on the latest revision (aaa60fae05f25f93010f917431db7fe462d9864c), it doesn't happen anymore; that error message isn't displayed.

However, the packages list is now empty and it doesn't display packages to be updated, even when there are packages to be updated on the system.

**Comment by danielhb** [comment]
I've sent a patch to the ML. Let me know if the problem is fixed

**Comment by alinefm** [comment]
Fixed by 5924897be6e39d67c229f53b68dcc21beb3bc184
